### PR TITLE
[CDAP-9088][CDAP-9327] Spark Program & Library

### PR DESCRIPTION
**Pull request by elfenheart**
JIRAs:
- https://issues.cask.co/browse/CDAP-9088
- https://issues.cask.co/browse/CDAP-9327

#### Changes
1. Add Spark program type to pipeline
2. Allow a pipeline that consists only 1 action
3. Add Library Resource Center Entity.

### CODE REVIEW

File: cdap-ui/app/cdap/text/text-en.yaml [hunk 1]
```diff
@@ -729,26 +734,48 @@ features:
...
     ArtifactUpload:
-      headerlabel: Add Third party Driver
+      headerlabel: Add Third Party Driver
       footertitle: Add Driver
       Step1:
         description: Upload your Driver
         shorttitle: Upload Driver
         title: Upload JAR
-        uploadHelperText: Upload the third party driver that was downloaded from the previous step
+        uploadHelperText: Upload the third party drvier that was downloaded from the previous step
```
Comment (tonybach) [inline cdap-ui/app/cdap/text/text-en.yaml]: typo

File: cdap-ui/app/cdap/text/text-en.yaml [hunk 2]
```diff
@@ -729,26 +734,48 @@ features:
...
         filePathLabel: Choose file
       Step2:
-        description: Configure the settings for your driver
+        description: Configure the settings for your drvier
```
Comment (tonybach) [inline cdap-ui/app/cdap/text/text-en.yaml]: here too

File: cdap-ui/app/cdap/components/CaskWizards/LibraryUpload/ConfigureStep/index.js [hunk 3]
```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
```
Comment (tonybach) [inline cdap-ui/app/cdap/components/CaskWizards/LibraryUpload/ConfigureStep/index.js]: Should be `2017`. Same in all the other newly created files.

File: cdap-ui/app/cdap/text/text-en.yaml [hunk 4]
```diff
@@ -729,26 +734,48 @@ features:
         filePathLabel: Choose file
       success: You have successfully deployed the application "{appName}".
       callToAction: View Application Details
+    LibraryUpload:
```
Comment (tonybach) [inline cdap-ui/app/cdap/text/text-en.yaml]: Should we arrange the Wizard types in order too?

**Review by tonybach — COMMENTED** [review]
minor comments

**Comment by ajainarayanan** [comment]
Talked to @elfenheart. Decided we could reuse the common steps from plugin upload wizard. We just need a different configure step.

**Comment by elfenheart** [comment]
@ajainarayanan so the upload wizard still have different i18n so we cannot reuse..

**Comment by bdmogal** [comment]
1. Type Placeholder: ``Library Type E.g. sparkprogram``
2. Class Name Placeholder:  ``Library Class. E.g. com.example.MyClass``
3. Driver Description: ``A JDBC driver is a JAR file that contains third-party code to communicate with relational databases, such as MySQL, Oracle, and PostgreSQL.

**Comment by bdmogal** [comment]
Other than that, LGTM

**Comment by bdmogal** [comment]
We need an icon for sparkprogram action type:

image

**Comment by bdmogal** [comment]
please check my update. Other than the missing icon, rest LGTM

**Comment by elfenheart** [comment]
Talked with @bdmogal, the plugin icon should be fix when we add the ability to add custom icon for any pipeline plugin.

**Comment by elfenheart** [comment]
Build: https://builds.cask.co/browse/CDAP-DRC6282-1

**Comment by tonybach** [comment]
Code LGTM 👍

**Pull request merged**